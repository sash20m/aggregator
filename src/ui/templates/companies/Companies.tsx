/* eslint-disable camelcase */
import React, { useState, useEffect } from 'react';

import { GetServerSidePropsContext } from 'next';
import { CompanyCard } from 'ui/organisms/companyCard/CompanyCard';
import { getCompanies } from 'services/companies.services';
import { HeaderBar } from 'ui/molecules/headerbar/Headerbar';
import { SearchInput } from 'ui/molecules/searchInput/SearchInput';
import { Pagination } from 'ui/molecules/pagination/Pagination';

import './CompaniesList.scss';

export interface CompaniesCard {
  name: string;
  slug: string;
  location: string;
  website: string | null;
  idno: string;
  employees: string;
  turnover: string;
  industry: string;
  mobile: boolean | null;
  phone: boolean | null;
  email: boolean | null;
  creation_year: number;
}

export interface Companies {
  data: CompaniesCard[];
  pages: number;
  total_results: number;
}

interface Props {
  data: Companies;
  companyName: string | string[] | undefined;
}

export const CompaniesList: React.FC<Props> = ({ companyName, data }) => {
  const [companies, setCompanies] = useState<Companies | null>();

  useEffect(() => {
    const onSetData = () => {
      setCompanies(data);
    };
    onSetData();
  }, [data]);

  const changePage = async (newPage: string) => {
    const companiesData = await getCompanies(companyName, newPage, 10);
    setCompanies(companiesData);
  };

  if (!companies) return <></>;

  return (
    <div className="layout-companies" data-testid="companies-list">
      <HeaderBar searchVisible={false} />
      <div className="layout-companies__search-area-small">
        <div className="layout-companies__search-wrapper">
          <SearchInput suggestions={null} />
        </div>
      </div>
      <div className="nr-results-layout">
        <img className="nr-results-layout__bag-case" src="../bag.png" alt="" />
        <p data-testid="total-results">{companies.total_results}</p>
        <p>of results</p>
      </div>
      {companies &&
        companies.data.map((companyInfo: CompaniesCard) => (
          <CompanyCard data={companyInfo} key={companyInfo.name} />
        ))}
      <Pagination pages={companies.pages} changePage={changePage} />
    </div>
  );
};

interface SSProps {
  props: Props;
}

export const getSSProps = async ({
  params,
}: GetServerSidePropsContext): Promise<SSProps | null> => {
  if (params) {
    const page = '1';
    const perPage = 10;
    const data = await getCompanies(params.slug, page, perPage);
    return {
      props: { data, companyName: params.slug },
    };
  }
  return null;
};
