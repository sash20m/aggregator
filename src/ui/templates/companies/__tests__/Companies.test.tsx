import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import { CompaniesList } from '../Companies';

const data = {
  data: [
    {
      name: 'Comapny Name',
      slug: 'aridon',
      location: 'Chisinau',
      website: 'https://ebs-integrator.com',
      idno: '123456789',
      employees: '10-20',
      turnover: '999999',
      industry: 'IT',
      mobile: true,
      phone: true,
      email: true,
      creation_year: 2020,
    },
  ],
  pages: 1,
  total_results: 1,
};

const companyName = 'aridon';

afterEach(cleanup);

describe('company ', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <CompaniesList data={data} companyName={companyName} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered', async () => {
    const { getByTestId } = render(
      <CompaniesList data={data} companyName={companyName} />
    );

    expect(getByTestId('companies-list')).toBeInTheDocument();
  });

  it('should have 2 children', async () => {
    const { getByTestId } = render(
      <CompaniesList data={data} companyName={companyName} />
    );

    expect(getByTestId('total-results')).toHaveTextContent('1');
  });
});
