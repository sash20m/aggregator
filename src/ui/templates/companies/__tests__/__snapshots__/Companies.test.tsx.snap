// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company  should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout-companies"
    data-testid="companies-list"
  >
    <div
      class="header-main"
      data-testid="headerbar"
    >
      <div
        class="header-main__content"
      >
        <button
          class="logo__btn"
          type="button"
        >
          <img
            alt="Logo"
            class="logo"
            src="../logo-small.png"
          />
        </button>
      </div>
    </div>
    <div
      class="layout-companies__search-area-small"
    >
      <div
        class="layout-companies__search-wrapper"
      >
        <div
          class="search-content"
        >
          <div
            class="search-content__input-wrapper"
          >
            <input
              class="search-content__input"
              data-testid="search-input"
              placeholder="Search from 225,195 companies"
              type="text"
              value=""
            />
            <img
              alt=""
              class="search-content__search-magnifier"
              src="../search.png"
            />
          </div>
          <div
            class="search-content__options"
          >
            <p
              class="search-content__options__item"
            >
              Search In
            </p>
            <p
              class="search-content__options__item--selected"
            >
              Companies
            </p>
            <p
              class="search-content__options__item"
            >
              Persons
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nr-results-layout"
    >
      <img
        alt=""
        class="nr-results-layout__bag-case"
        src="../bag.png"
      />
      <p
        data-testid="total-results"
      >
        1
      </p>
      <p>
        of results
      </p>
    </div>
    <div
      class="results"
    >
      <div
        class="results__item"
      >
        <button
          class="results__item__header"
          data-testid="aridon"
          type="button"
        >
          <img
            alt=""
            class="results__item__header__avatar"
            src="https://account.globaldatabase.com/logo/ebs-integrator.com/"
          />
          <div>
            <div
              class="results__item__header__name"
            >
              Comapny Name
            </div>
            <div
              class="results__item__header__location"
            >
              Chisinau
            </div>
          </div>
        </button>
        <div
          class="results__item__company-info"
        >
          <div
            class="results__item__company-info__column"
          >
            <div
              class="results__item__company-info__column__item"
            >
              <p
                class="results__item__company-info__column__item--key"
              >
                IDNO:
              </p>
              <p
                class="results__item__company-info__column__item--value"
              >
                123456789
              </p>
            </div>
            <div
              class="results__item__company-info__column__item"
            >
              <p
                class="results__item__company-info__column__item--key"
              >
                Status
              </p>
              <button
                class="results__item__company-info__column__item--value"
                type="button"
              >
                ACTIV
              </button>
            </div>
            <div
              class="results__item__company-info__column__item"
            >
              <p
                class="results__item__company-info__column__item--key"
              >
                Date of establishment:
              </p>
              <p
                class="results__item__company-info__column__item--value"
              >
                2020
              </p>
            </div>
            <div
              class="results__item__company-info__column__item"
            >
              <p
                class="results__item__company-info__column__item--key"
              >
                Varsta
              </p>
              <p
                class="results__item__company-info__column__item--value"
              >
                0
              </p>
            </div>
          </div>
          <div
            class="results__item__company-info__column"
          >
            <div
              class="results__item__company-info__column__item"
            >
              <p
                class="results__item__company-info__column__item--key"
              >
                Nr. by the employees:
              </p>
              <p
                class="results__item__company-info__column__item--value"
              >
                10-20
              </p>
            </div>
            <div
              class="results__item__company-info__column__item"
            >
              <p
                class="results__item__company-info__column__item--key"
              >
                Turnover:
              </p>
              <p
                class="results__item__company-info__column__item--value"
              >
                999999 MDL
              </p>
            </div>
            <div
              class="results__item__company-info__column__item"
            >
              <p
                class="results__item__company-info__column__item--key"
              >
                Industry
              </p>
              <p
                class="results__item__company-info__column__item--value"
              >
                IT
              </p>
            </div>
          </div>
          <div
            class="results__item__company-info__column--row"
          >
            <div
              class="results__item__company-info__column__item--key"
            >
              Contact
            </div>
            <div>
              <div
                class="results__item__company-info__column__item--value"
              >
                <div
                  class="results__item__company-info__column__item__results"
                >
                  <img
                    alt="phone"
                    class="results__item__company-info__column__item__icon"
                    src="../phone.png"
                  />
                  <p>
                    Phone
                  </p>
                </div>
              </div>
              <div
                class="results__item__company-info__column__item--value"
              >
                <div
                  class="results__item__company-info__column__item__results"
                >
                  <img
                    alt="phone"
                    class="results__item__company-info__column__item__icon"
                    src="../mobile.png"
                  />
                  <p>
                    Mobile
                  </p>
                </div>
              </div>
              <div
                class="results__item__company-info__column__item--value"
              >
                <div
                  class="results__item__company-info__column__item__results"
                >
                  <img
                    alt="phone"
                    class="results__item__company-info__column__item__icon"
                    src="../email.png"
                  />
                  <p>
                    Email
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination"
    >
      <button
        class="pagination__arrow-btn"
        data-testid="goOnePageBack"
        type="button"
      >
        &lt;
      </button>
      <button
        class="pagination__page-btn--active"
        data-testid="0"
        name="1"
        type="button"
      >
        1
      </button>
      <button
        class="pagination__arrow-btn"
        data-testid="goOnePageFurther"
        type="button"
      >
        &gt;
      </button>
    </div>
  </div>
</DocumentFragment>
`;
